src/mspdb.cpp

```diff
@@ -165,6 +165,94 @@ bool tryLoadMsPdbVS2017(const char* mspdb, const char* path = 0)
 	return modMsPdb != 0;
 }
 
+bool tryLoadMsPdbVSWhere(const char* mspdb)
+{
+	HANDLE read;
+	HANDLE write;
+
+	SECURITY_ATTRIBUTES securityAttributes = { 0 };
+	securityAttributes.nLength = sizeof(SECURITY_ATTRIBUTES);
+	securityAttributes.lpSecurityDescriptor = NULL;
+	securityAttributes.bInheritHandle = TRUE;
+	if(!CreatePipe(&read, &write, &securityAttributes, 0) ||
+	   !SetHandleInformation(read, HANDLE_FLAG_INHERIT, 0))
+		return false;
+
+	STARTUPINFOA startupInfo = { 0 };
+	startupInfo.cb = sizeof(startupInfo);
+	startupInfo.dwFlags = STARTF_USESTDHANDLES;
+	startupInfo.hStdInput = GetStdHandle(STD_INPUT_HANDLE);
+	startupInfo.hStdOutput = write;
+	startupInfo.hStdError = GetStdHandle(STD_ERROR_HANDLE);
+	PROCESS_INFORMATION processInformation = { 0 };
+	ZeroMemory(&processInformation, sizeof(PROCESS_INFORMATION));
+
+	char vsPath[260] = "";
+	char commandLine[] = "vswhere.exe -latest -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath";
+	if(CreateProcessA("C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\vswhere.exe", commandLine, NULL, NULL, TRUE, 0, NULL, NULL, &startupInfo, &processInformation))
+	{
+		CloseHandle(processInformation.hProcess);
+		CloseHandle(processInformation.hThread);
+
+		DWORD length;
+		if(ReadFile(read, vsPath, sizeof(vsPath), &length, NULL) && length > 2)
+		{
+			// trim trailing CR/LF
+			if (vsPath[length - 1] == '\n')
+			{
+				vsPath[--length] = '\0';
+				if (vsPath[length - 1] == '\r')
+					vsPath[--length] = '\0';
+			}
+		}
+	}
+	CloseHandle(read);
+	CloseHandle(write);
+	if (vsPath[0] == '\0')
+		return false;
+
+	strncat(vsPath, "\\VC\\Tools\\MSVC\\*", 260);

```

Comment:
Instead of searching the directory, the direct way is to read the installed/desired version from VC\Auxiliary\Build\Microsoft.VCToolsVersion.default.txt.

Reply:
Excellent, thank you! I had no idea that this file existed.